tests/php/FileSessionHandlerTest.php

```diff
@@ -0,0 +1,720 @@
+                'sessionSavePath' => static::$sessionSavePath,
+                'baseDir' => static::$sessionSavePath,
+                'numSubDirs' => 0,
+                'mode' => 0600,
+            ],
+            'path and subdirs' => [
+                'sessionSavePath' => '1;' . static::$sessionSavePath,
+                'baseDir' => static::$sessionSavePath,
+                'numSubDirs' => 1,
+                'mode' => 0600,
+            ],
+            'all config defined' => [
+                'sessionSavePath' => '2;123;' . static::$sessionSavePath,
+                'baseDir' => static::$sessionSavePath,
+                'numSubDirs' => 2,
+                'mode' => 0123,
+            ],
+            'full octal mode' => [
+                'sessionSavePath' => '3;0766;' . static::$sessionSavePath,
+                'baseDir' => static::$sessionSavePath,
+                'numSubDirs' => 3,
+                'mode' => 0766,
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider provideSetSavePath
+     */
+    public function testSetSavePath(string $sessionSavePath, string $baseDir, int $numSubDirs, int $mode): void
+    {
+        // Set path
+        $handler = new FileSessionHandler();
+        $reflectionMethod = new ReflectionMethod($handler, 'setSavePath');
+        $reflectionMethod->setAccessible(true);
+        $reflectionMethod->invoke($handler, $sessionSavePath);
+
+        // Check baseDir
+        $reflectionBaseDir = new ReflectionProperty($handler, 'baseDir');
+        $reflectionBaseDir->setAccessible(true);
+        $this->assertSame($baseDir, $reflectionBaseDir->getValue($handler));
+
+        // Check numSubDirs
+        $reflectionNumSubDirs = new ReflectionProperty($handler, 'numSubDirs');
+        $reflectionNumSubDirs->setAccessible(true);
+        $this->assertSame($numSubDirs, $reflectionNumSubDirs->getValue($handler));
+
+        // Check mode
+        $reflectionMode = new ReflectionProperty($handler, 'mode');
+        $reflectionMode->setAccessible(true);
+        $this->assertSame($mode, $reflectionMode->getValue($handler));
+    }
+
+    public static function provideGetFilePath(): array
+    {
+        return [
+            'path no subdirs' => [
+                'savePath' => static::$sessionSavePath,
+                'sessionID' => 'a0f123456789',
+                'expected' => Path::join(static::$sessionSavePath, 'sess_a0f123456789'),
+            ],
+            'path three subdirs' => [
+                'savePath' => '3;' . static::$sessionSavePath,
+                'sessionID' => 'a0f123456789',
+                // Should be `a/0/f` but there's a bug in the Path::join algorithm.
+                // We fixed that in CMS 6 but decided it was a bit risky to fix in CMS 5.
+                // The end result is still the same, i.e. the file is still saved and found correctly.
+                'expected' => Path::join(static::$sessionSavePath, 'a/f', 'sess_a0f123456789'),
```

Comment:
Had to change this expected value.
There's no risk of collisions (the filename itself isn't affected) and there's no reduction in functionality with the one mega-edge-case that an existing session file that the baked-in PHP session handler made won't be found at `a/0/f/sess_a0f123456789` when the new session handler becomes active, so a new session will open instead. 

Which is completely fine. Worst case, a single user gets logged out temporarily one time after the initial deployment.